packages/quantic/force-app/main/default/lwc/quanticTimeframeFacet/quanticTimeframeFacet.js

```diff
@@ -0,0 +1,526 @@
+import {
+  getHeadlessBindings,
+  initializeWithHeadless,
+  registerComponentForInit,
+} from 'c/quanticHeadlessLoader';
+import {
+  fromLocalIsoDate,
+  fromSearchApiDate,
+  I18nUtils,
+  RelativeDateFormatter,
+  toLocalIsoDate,
+  toLocalSearchApiDate,
+} from 'c/quanticUtils';
+import {api, LightningElement, track} from 'lwc';
+
+import expandFacet from '@salesforce/label/c.quantic_ExpandFacet';
+import collapseFacet from '@salesforce/label/c.quantic_CollapseFacet';
+import clearFilter from '@salesforce/label/c.quantic_ClearFilter';
+import startLabel from '@salesforce/label/c.quantic_StartLabel';
+import endLabel from '@salesforce/label/c.quantic_EndLabel';
+import apply from '@salesforce/label/c.quantic_Apply';
+
+/** @typedef {import("coveo").SearchEngine} SearchEngine */
+/** @typedef {import("coveo").SearchStatus} SearchStatus */
+/** @typedef {import("coveo").DateFacet} DateFacet */
+/** @typedef {import("coveo").DateFacetState} DateFacetState */
+/** @typedef {import("coveo").DateFilter} DateFilter */
+/** @typedef {import("coveo").DateFilterState} DateFilterState */
+/** @typedef {import("coveo").DateRangeRequest} DateRangeRequest */
+/** @typedef {import("coveo").DateFacetValue} DateFacetValue */
+/** @typedef {import("coveo").RelativeDatePeriod} RelativeDatePeriod */
+/** @typedef {import("coveo").RelativeDateUnit} RelativeDateUnit */
+/**
+ * @typedef {Object} DatepickerElement
+ * @property {string} min
+ * @property {string} max
+ * @property {string} value
+ * @property {boolean} required
+ * @property {Function} reportValidity
+ */
+/**
+ * @typedef {Object} TimeframeElement
+ * @property {RelativeDatePeriod} period
+ * @property {RelativeDateUnit} unit
+ * @property {(string|number)} amount
+ * @property {string} label
+ */
+/**
+ * @typedef {Object} Timeframe
+ * @property {RelativeDatePeriod} period
+ * @property {RelativeDateUnit} unit
+ * @property {number} amount
+ * @property {string} label
+ */
+
+/**
+ * The `QuanticTimeframeFacet` component displays dates as relative ranges.
+ * 
```

Comment:
might be an extra unecessary newline here. For separating paragraphs in the description two newlines are necessary but the @example tag breaks the description text here anyway.